src/Conversion/ONNXToTorch/NN/AddOp.cpp

```diff
@@ -0,0 +1,134 @@
+using namespace mlir::torch::Torch;
+
+/**
+ *
+ * ONNX Add operation
+ *
+ * “Performs element-wise binary addition (with Numpy-style broadcasting
+ * support).” “” “This operator supports multidirectional
+ * (i.e. Numpy-style) broadcasting; for more details please check the doc.”
+ *
+ * Operands :
+ *    A	  tensor of 32-bit/64-bit unsigned integer values or
+ *        tensor of 32-bit/64-bit signless integer values or
+ *        tensor of 16-bit/32-bit/64-bit float values or
+ *        tensor of bfloat16 type values or memref of any type values
+ *  Map this A operand with input parameter in torch side.
+ *    B   tensor of 32-bit/64-bit unsigned integer values or
+ *        tensor of 32-bit/64-bit signless integer values or
+ *        tensor of 16-bit/32-bit/64-bit float values or
+ *        tensor of bfloat16 type values or memref of any type values
+ *  Map this B operand with other parameter in torch side.
+ *
+ * Results:
+ *   C    tensor of 32-bit/64-bit unsigned integer values or
+ *        tensor of 32-bit/64-bit signless integer values or
+ *        tensor of 16-bit/32-bit/64-bit float values or
+ *        tensor of bfloat16 type values or memref of any type values
+ *
+ */
+
+class ONNXAddOpToTorchLowering : public ConversionPattern {
+public:
+  ONNXAddOpToTorchLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(
+            typeConverter, mlir::ONNXAddOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op->getLoc();
+    mlir::MLIRContext *context = op->getContext();
+    ONNXAddOp op1 = llvm::dyn_cast<ONNXAddOp>(op);
+    ONNXAddOpAdaptor adapter(op1);
+
+    auto a = op1.A();
+    auto b = op1.B();
+    TensorType a_tensor_type = a.getType().cast<TensorType>();
+    TensorType b_tensor_type = b.getType().cast<TensorType>();
+    TensorType op_tensor_type =
+	    op->getResult(0).getType().cast<TensorType>();
+    auto ty = IntegerType::get(op1.getContext(), 64);
+    auto one = 1;
+    auto f1 = IntegerAttr::get(ty, one);
+    Value f1v = rewriter.create<ConstantIntOp>(loc, f1);
+
+    auto aTy = Torch::ValueTensorType::get(
+        context, a_tensor_type.getShape(), a_tensor_type.getElementType());
+    auto att = rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>
+	    (loc, aTy, a);
+    auto bTy = Torch::ValueTensorType::get(context,
+		b_tensor_type.getShape(), b_tensor_type.getElementType());
+    auto btt = rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>
+	    (loc, bTy, b);
+    auto resultTy = Torch::ValueTensorType::get(op1.getContext(),
+        op_tensor_type.getShape(), op_tensor_type.getElementType());
+    Value atenaddresult =
+        rewriter.create<AtenAddTensorOp>(loc, resultTy, att, btt, f1v);
+
+    llvm::outs() << "ATENADDOP CREATED is " << atenaddresult << "\n";
+    Value result = atenaddresult;
```

Comment:
NIT: this doesn't seem to add any information. Is it here because of a Helprack coding guideline? If not, it would be easier to read without this additional copy.

Reply:
Resolved